packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/index.js

```diff
@@ -81,14 +82,23 @@ function shimSmithySend(originalSend) {
     const command = smithySendArgs[0];
 
     if (isActive) {
-      const awsProduct = operationMap[command.constructor.name];
+      const serviceId = self.config && self.config.serviceId;
+      const awsProduct = serviceId && awsProducts.find(aws => aws.spanName === serviceId.toLowerCase());
 
-      if (awsProduct) {
+      if (awsProduct && awsProduct.supportsOperation(command.constructor.name)) {
         return awsProduct.instrumentedSmithySend(self, originalSend, smithySendArgs);
+      } else {
+        // This code will be removed when we release support for version 3x.
+        // We are keeping it here for now to avoid any unknown problems.
```

Comment:
_question:_ I do not understand this comment. Does the "support for 3.x" mean "we will remove this when we release @instana/core@3.x? Or does the 3.x refer to the version of the AWS SDK or the kinesis client? 

AFAICT, we can only remove this, when we have migrated all other AWS SDK v3 instrumentations to this new pattern, right?

Reply:
Absolutely! Our current plan is to remove the 'else' section within version 3.x. We're optimistic that by that time, we'll have successfully transitioned all other AWS SDK v3 instrumentations to this new approach. [ref](https://instana.kanbanize.com/ctrl_board/56/cards/96212/details/). @kirrg001